Mage.Tests/src/test/java/org/mage/test/cards/single/bro/OneWithTheMultiverseTest.java

```diff
@@ -0,0 +1,370 @@
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, ogre, true);
+        setChoice(playerA, owtm);
+        checkPlayableAbility("can cast for free", 1, PhaseStep.PRECOMBAT_MAIN, playerA, "Cast Gray Ogre", true);
+        checkPlayableAbility("can cast for free", 1, PhaseStep.PRECOMBAT_MAIN, playerA, "Cast Goblin Piker", true);
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, ogre, true);
+        setChoice(playerA, "Without paying manacost");
+        checkPlayableAbility("can't cast", 1, PhaseStep.PRECOMBAT_MAIN, playerA, "Cast Gray Ogre", false);
+        checkPlayableAbility("can't cast", 1, PhaseStep.PRECOMBAT_MAIN, playerA, "Cast Goblin Piker", false);
+
+        setStopAt(1, PhaseStep.PRECOMBAT_MAIN);
+        execute();
+
+        assertPermanentCount(playerA, ogre, 2);
+        assertPermanentCount(playerA, piker, 0);
+        assertLibraryCount(playerA, ogre, 1);
+        assertHandCount(playerA, piker, 2);
+        assertTappedCount("Volcanic Island", true, 8 + 3);
+    }
+
+    @Test
+    public void castNormalFromHandThenFreeFromHand() {
+        setStrictChooseMode(true);
+        skipInitShuffling();
+
+        // The "You may look at the top card of your library any time."
+        // is not set up properly if starting directly on the battlefield.
+        // So we do cast it in those tests.
+        addCard(Zone.HAND, playerA, owtm);
+        addCard(Zone.BATTLEFIELD, playerA, "Volcanic Island", 11);
+
+        addCard(Zone.LIBRARY, playerA, ogre, 3);
+        addCard(Zone.HAND, playerA, piker, 2);
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, owtm, true);
+        checkPlayableAbility("can cast for free", 1, PhaseStep.PRECOMBAT_MAIN, playerA, "Cast Gray Ogre", true);
+        checkPlayableAbility("can cast for free", 1, PhaseStep.PRECOMBAT_MAIN, playerA, "Cast Goblin Piker", true);
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, piker, true);
+        setChoice(playerA, piker);
+        checkPlayableAbility("can cast for free", 1, PhaseStep.PRECOMBAT_MAIN, playerA, "Cast Gray Ogre", true);
+        checkPlayableAbility("can cast for free", 1, PhaseStep.PRECOMBAT_MAIN, playerA, "Cast Goblin Piker", true);
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, piker, true);
+        setChoice(playerA, "Without paying manacost");
+        checkPlayableAbility("can't cast", 1, PhaseStep.PRECOMBAT_MAIN, playerA, "Cast Gray Ogre", false);
+        checkPlayableAbility("can't cast", 1, PhaseStep.PRECOMBAT_MAIN, playerA, "Cast Goblin Piker", false);
+
+        setStopAt(1, PhaseStep.PRECOMBAT_MAIN);
+        execute();
+
+        assertPermanentCount(playerA, ogre, 0);
+        assertPermanentCount(playerA, piker, 2);
+        assertLibraryCount(playerA, ogre, 3);
+        assertHandCount(playerA, piker, 0);
+        assertTappedCount("Volcanic Island", true, 8 + 2);
+    }
+
+    @Test
+    public void castNormalFromHandThenFreeFromTop() {
```

Comment:
(for future, there's probably a way to parametrize tests like these)

Reply:
Difference between them was subtle enough I did not want to make them using shared methods. Sorry if that made the review work harder.